### PR TITLE
feat(communities): add delete community channel feature

### LINKED ISSUE #2734: Add Delete to community channels
In terms of Edit channels feature, there should be a Delete channel functionality that was not implemented yet

https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=5203%3A43695

image

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #2734

Do not merge until the status-go PR is merged: https://github.com/status-im/status-go/pull/2266

### CODE REVIEW

File: src/app/chat/views/community_list.nim [hunk 1]
```diff
@@ -173,6 +173,14 @@ QtObject:
       let channelIdx = community.chats.findIndexById(channel.id)
       if channelIdx > -1:
         community.chats[channelIdx] = channel
+ 
+  proc removeChannelInCommunity*(self: CommunityList, communityId: string, channelId: string) =
+    var community = self.getCommunityById(communityId)
+    let idx = community.chats.findIndexById(channelId)
+    if idx == -1: return
+    community.chats.delete(idx)
```
Comment (richard-ramos) [inline src/app/chat/views/community_list.nim:181]: Is the message and pinned message list for the community deleted as well?

**Comment by anastasiyaig** [comment]
@jrainville 

for some reason, clear history stopped working for community member (not admin) and it works in master ( from last friday). Could u please take a look if this is something related?

https://images.zenhubusercontent.com/261270348/5fd9f102-8ec9-4311-839a-e5c2a90cd8ed/screen_recording_2021_06_28_at_11_46_36.mov

**Comment by jrainville** [comment]
@anastasiyaig I tried and it works for me. Did you have any error in the console?

**Comment by anastasiyaig** [comment]
nope, no errors in console, just options that do nothing. I will take a look again today

**Comment by anastasiyaig** [comment]
@jrainville indeed, I can't reproduce it now :( okay lets merge it

**Comment by jrainville** [comment]
Rebased

**Comment by jrainville** [comment]
The status-go PR was merge and this branch was updated to point to develop. It can now be merged

**Pull request merged**